railties/lib/rails/application/configuration.rb

```diff
@@ -649,6 +650,11 @@ def generate_local_secret
 
           File.binread(key_file)
         end
+
+        def generate_local_secret?
+          Rails.env.local? && !root.join("config/credentials/#{Rails.env}.yml.enc").exist? ||
+            ENV["SECRET_KEY_BASE_DUMMY"]
```

Comment:
`secret_key_base` might not be defined in credentials:
```suggestion
          Rails.env.local? && !Rails.application.credentials.secret_key_base ||
            ENV["SECRET_KEY_BASE_DUMMY"]
```

Also, this will generate a local secret in production if `SECRET_KEY_BASE_DUMMY` is set:
```
false && false || true
=> true
```

```suggestion
          Rails.env.local? &&
            (!Rails.application.credentials.secret_key_base ||
             ENV["SECRET_KEY_BASE_DUMMY"])
```

Reply:
Agree. Can someone update this patch so it cover this case as well?